### PR TITLE
print the error sent from the server

### PR DESCRIPTION
**Pull request by chuenlok**
# What does this PR do?

The assert errors from the server are not returned to the client and hid by the 500 internal server error. Updated the server code (https://github.com/meta-llama/llama-stack/pull/493) to send the error and this client can print the error.

- [x] Addresses issue (#434)

## Test Plan

Please describe:
 - tests you ran to verify your changes with result summaries.
 - provide instructions so it can be reproduced.

1. Kick off the server with `llama stack run /Users/henrytai/.llama/distributions/llamastack-ollama/ollama-run.yaml --port 5050`
1. Trigger an error case from client. The example is `PYTHONPATH=. python -m examples.agents.client localhost 5050` from the `llama-stack-apps` repo. The issue details:  "This is a known issue with Llama3.2-3B-Instruct on ollama, where the model produces bad tool calling outputs. In this case, the server agent fails due to assert error that "code_interpreter" is not in the AgentConfig's tools field" but the code_interpreter is not really necessary as it is not needed in Llama3.1-8B.

client log:
```
inference> assistant<|end_header_id|>

Switzerland is a beautiful country with a rich history, stunning mountains, and vibrant cities. Here are three must-visit places in Switzerland:

1. **Geneva**: Geneva is a global hub for international diplomacy and is often referred to as the "City of Peace". It's home to ...

These are just a few examples of Switzerland's amazing attractions!
500: Internal server error: An unexpected error occurred. Tool code_interpreter not found
inference> brave_search.call(query="what is special about #1")...
```
Showing the real error in the client: **Tool code_interpreter not found**

## Sources

Please link relevant resources if necessary.

## Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Ran pre-commit to handle lint / formatting issues.
- [X] Read the [contributor guideline](https://github.com/meta-llama/llama-stack/blob/main/CONTRIBUTING.md),
      Pull Request section?
- [ ] Updated relevant documentation.
- [ ] Wrote necessary unit or integration tests.

### CODE REVIEW

**Comment by chuenlok** [comment]
@yanxi0830 Can you help review this?

**Comment by yanxi0830** [comment]
Thanks! Could you rebase the change on top of current main? There have been some refactors done on the agents lib.

File: src/llama_stack_client/lib/agents/agent.py [hunk 1]
```diff
@@ -51,7 +57,10 @@ async def create_turn(
         )
         turn = None
         for chunk in response:
-            if chunk.event.payload.event_type != "turn_complete":
+            if hasattr(chunk, "error"):
```
Comment (yanxi0830) [inline src/llama_stack_client/lib/agents/agent.py:60]: What is the chunk's value that will be returned if an error happens?

Comment (ashwinb) [reply src/llama_stack_client/lib/agents/agent.py:60]: yesh seems like we return either
- a chunk
- a message (for tool call) ! 
- and now, an error dict!

**Comment by chuenlok** [comment]
> Thanks! Left some questions. Could you rebase the change on top of current main? There have been some refactors done on the agents lib.

Rebased!

File: src/llama_stack_client/lib/agents/event_logger.py [hunk 2]
```diff
@@ -153,4 +159,4 @@ def log(self, event_generator):
         for chunk in event_generator:
             for log_event in self._get_log_event(chunk, previous_event_type, previous_step_type):
                 yield log_event
-            previous_event_type, previous_step_type = self._get_event_type_step_type(chunk)
\ No newline at end of file
+            previous_event_type, previous_step_type = self._get_event_type_step_type(chunk)
```
Comment (yanxi0830) [inline src/llama_stack_client/lib/agents/event_logger.py:162]: Could you install and run pre-commit? `pre-commit run --all-files `

**Pull request merged**